packages/repository/examples/models/order.ts

```diff
@@ -11,14 +11,14 @@ import {Customer} from './customer';
 @model()
 class Order extends Entity {
   @property({
-    name: 'qty',
+    type: 'number',
```

Comment:
The `type` should be optional for TypeScript based property as it can be inferred.

Reply:
Good point. 

AFAIK, the reflection-based type inference provides only values that you can use with `typeof` operator, typically `string`, `number`, `boolean`, `object` or `function` (see [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof)). For example, a `Date` type is considered as an `object`. A model/entity constructor is considered as an `function`.

IMO, this make property type inference so limited that it's actually useless. I am also concerned that people will not be aware of these limitation, they will think LoopBack is able to infer the type (e.g. Date, Model constructor, etc.) when in fact we will use only `object` or `function`. (I guess this can be addressed by changing the code inferring property type to recognize only "string", "number", "boolean" and throw an error in all other cases.)

Having said that, the current version of `@loopback/authentication` does not support inference of property types. I am proposing to leave that out of this pull request.

Reply:
Add a comment in 878ec6b